R/preprocess.R

```diff
@@ -193,20 +258,61 @@ enw_new_reports <- function(obs) {
     by = c("group")
   ]
   reports <- reports[, delay := 0:(.N - 1), by = c("reference_date", "group")]
+
+  if (!is.null(reports$max_confirm)) {
+    reports[, prop_reported := new_confirm / max_confirm]
+  }
+
+  if (set_negatives_to_zero) {
+    reports <- reports[new_confirm < 0, new_confirm := 0]
```

Comment:
Will lead to inconsistencies between `new_confirm` and `prop_reported`. Could be better to have them synchronized, i.e. set `prop_reported` to zero as well..

Another thing to note is that max_confirm will currently be smaller than the sum of `new_confirm` if there are negative entries. Not sure if this is an issue, and what is more important: internal consistency in the dataframe vs. reminding the user that this data has been manipulated by having the original max_confirm remain as it was

Reply:
Yeah, this is a good point and as in #52 we need to be more nuanced with what we are doing here. 

So there is a tension between cleaning the data as we need and highlighting we have done so to the user. Perhaps we should be making a new variable (i.e `new_confirm_adj`) and using that to adjust the data etc? 

Otherwise I agree moving setting negatives to zero above calculating the proportion reported makes sense. Also agree that leaving max_confirm as is makes sense to flag to the user.